packages/language-server/src/plugins/typescript/language-service.ts

```diff
@@ -223,8 +223,22 @@ async function createLanguageService(
 
 		astroModuleLoader.deleteUnresolvedResolutionsFromCache(fileName);
 		doc = DocumentSnapshotUtils.createFromFilePath(fileName, docContext.createDocument);
+
 		snapshotManager.set(fileName, doc);
 
+		// If we needed to create an Astro snapshot, also create its script tags snapshots
+		if (isAstroFilePath(fileName)) {
+			const document = (doc as AstroSnapshot).parent;
+
+			document.scriptTags.forEach((scriptTag, index) => {
+				const scriptFilePath = fileName + `.__script${index}.js`;
```

Comment:
See this being done in a couple of places now. By the rule of three it would be good to move into a helper function if it's needed again in the future.